lib/modem_info/modem_info.c

```diff
@@ -159,32 +197,49 @@ static const struct modem_info_data iccid_data = {
 	.data_type = AT_PARAM_TYPE_STRING,
 };
 
+static const struct modem_info_data systemmode_data = {
+	.cmd = AT_CMD_SYSTEMMODE,
+	.param_index = SYSTEMMODE_PARAM_INDEX,
+	.param_count = SYSTEMMODE_PARAM_COUNT,
+	.data_type = AT_PARAM_TYPE_NUM_SHORT,
+};
+
 static const struct modem_info_data *const modem_data[] = {
-	[MODEM_INFO_RSRP] = &rsrp_data,
-	[MODEM_INFO_BAND] = &band_data,
-	[MODEM_INFO_MODE] = &mode_data,
-	[MODEM_INFO_OPERATOR] = &operator_data,
-	[MODEM_INFO_CELLID] = &cellid_data,
-	[MODEM_INFO_IP_ADDRESS] = &ip_data,
-	[MODEM_INFO_UICC] = &uicc_data,
-	[MODEM_INFO_BATTERY] = &battery_data,
-	[MODEM_INFO_TEMP] = &temp_data,
-	[MODEM_INFO_FW_VERSION] = &fw_data,
-	[MODEM_INFO_ICCID] = &iccid_data,
+	[MODEM_INFO_RSRP]	= &rsrp_data,
+	[MODEM_INFO_CUR_BAND]	= &band_data,
+	[MODEM_INFO_SUP_BAND]	= &band_sup_data,
+	[MODEM_INFO_MODE]	= &mode_data,
+	[MODEM_INFO_OPERATOR]	= &operator_data,
+	[MODEM_INFO_MCC]	= &mcc_data,
+	[MODEM_INFO_MNC]	= &mnc_data,
+	[MODEM_INFO_AREA_CODE]	= &area_data,
+	[MODEM_INFO_CELLID]	= &cellid_data,
+	[MODEM_INFO_IP_ADDRESS]	= &ip_data,
+	[MODEM_INFO_UICC]	= &uicc_data,
+	[MODEM_INFO_BATTERY]	= &battery_data,
+	[MODEM_INFO_TEMP]	= &temp_data,
+	[MODEM_INFO_FW_VERSION]	= &fw_data,
+	[MODEM_INFO_ICCID]	= &iccid_data,
+	[MODEM_INFO_SYSTEMMODE]	= &systemmode_data,
 };
 
 static const char *const modem_data_name[] = {
-	[MODEM_INFO_RSRP] = "RSRP",
-	[MODEM_INFO_BAND] = "BAND",
-	[MODEM_INFO_MODE] = "MODE",
-	[MODEM_INFO_OPERATOR] = "OPERATOR",
-	[MODEM_INFO_CELLID] = "CELLID",
-	[MODEM_INFO_IP_ADDRESS] = "IP ADDRESS",
-	[MODEM_INFO_UICC] = "UICC STATE",
-	[MODEM_INFO_BATTERY] = "BATTERY",
-	[MODEM_INFO_TEMP] = "TEMP",
-	[MODEM_INFO_FW_VERSION] = "MODEM FW",
-	[MODEM_INFO_ICCID] = "ICCID",
+	[MODEM_INFO_RSRP]	= "signalStrength",
```

Comment:
Why don't just put the name inside the modem_data object? Then you would not need to worry about the arrays getting out of sync.

Reply:
Fixed.